edgeless_node/src/agent/mod.rs

```diff
@@ -39,23 +39,25 @@ impl Agent {
         let (sender, receiver) = futures::channel::mpsc::unbounded();
 
         let main_task = Box::pin(async move {
-            Self::main_task(receiver, runner, resources, data_plane_provider).await;
+            Self::main_task(receiver, runners, resources, data_plane_provider).await;
         });
 
         (Agent { sender, node_settings }, main_task)
     }
 
     async fn main_task(
         receiver: futures::channel::mpsc::UnboundedReceiver<AgentRequest>,
-        runner: Box<dyn crate::base_runtime::RuntimeAPI + Send>,
+        mut runners: std::collections::HashMap<String, Box<dyn crate::base_runtime::RuntimeAPI + Send>>,
         resources: std::collections::HashMap<
             String,
             Box<dyn edgeless_api::resource_configuration::ResourceConfigurationAPI<edgeless_api::function_instance::InstanceId>>,
         >,
         data_plane_provider: edgeless_dataplane::handle::DataplaneProvider,
     ) {
         let mut receiver = std::pin::pin!(receiver);
-        let mut runner = runner;
+        // key: class_type
+        // value: RuntimeBox
+        //let mut runners = runners;
```

Comment:
This is probably left from some earlier version.
You can move the description to the function's interface (or add a type alias) and get rid of the let.

Reply:
Moved the key and value information to the parameters of `main_task`. Removed line 60 (let).